### PR TITLE
chore: add govulncheck workflow

### PR DESCRIPTION
**Pull request by shiiyan**
## Description

Closes https://github.com/sivchari/govalid/issues/105

Changes include:

* Runs govulncheck on every pull request, push to main, and on a weekly schedule.
* Checks out your code before scanning.
* Runs govulncheck using the Go version from go.mod file, scanning all packages.
* Saves the scan results to govulncheck.log.
* Uploads the report as an artifact, even if vulnerabilities are found and the job fails.

image

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [x] Refactoring (no functional changes)

## Quality Checklist

Before submitting this pull request, ensure your contribution meets these quality standards:

### Code Quality
- [ ] Code passes `make golangci-lint` without errors
- [ ] All tests pass: `make test`
- [ ] Code follows existing patterns and conventions
- [ ] No hard-coded values (use constants or configuration)
- [ ] Error messages are clear and actionable

### Testing Requirements
- [ ] Golden tests created and passing (`internal/analyzers/govalid/testdata/`)
- [ ] Unit tests implemented with boundary value testing (`test/unit/`)
- [ ] Benchmark tests comparing against popular validation libraries (`test/benchmark/`)
  - [ ] [go-playground/validator](https://github.com/go-playground/validator) (BenchmarkGoPlayground*)
  - [ ] [asaskevich/govalidator](https://github.com/asaskevich/govalidator) (BenchmarkGoValidator*)
  - [ ] [gookit/validate](https://github.com/gookit/validate) (BenchmarkGookitValidate*)
- [ ] All test files generated: `cd test && go generate`

### Performance Standards
- [ ] Zero allocations in validation logic (verified via benchmarks)
- [ ] Performance improvement over existing validation libraries (minimum 2x faster)
- [ ] Benchmark results added to `test/benchmark/README.md`

### Documentation
- [ ] README.md updated with new marker documentation
- [ ] Code includes appropriate comments and examples
- [ ] Validator usage examples provided

### Integration
- [ ] Validator scaffold generated with `make generate-validator MARKER=yourmarker`
- [ ] Registry files automatically updated
- [ ] Binary rebuilt and installed: `go install ./cmd/govalid/`

### Final Verification
- [x] All GitHub Actions CI checks pass
- [x] No breaking changes to existing functionality
- [x] Contribution follows project license requirements

### CODE REVIEW

**Comment by shiiyan** [comment]
@sivchari 

Please have a look when you get time.

File: .github/workflows/go-vulncheck.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+name: Vulnerability Check
+
+on:
+  pull_request:
```
[the remark this answers is not in the document]
Comment (shiiyan) [reply .github/workflows/go-vulncheck.yaml]: Fixed. Please check.

[.github/workflows/go-vulncheck.yaml](https://github.com/sivchari/govalid/pull/146#diff-5c42d8f230978fa9e40106e9143776f97240c10f2aed4f30523b71cf3d7d6c4e)

**Review by sivchari — APPROVED** [review]

**Comment by sivchari** [comment]
Thx!
